crates/bevy_ecs/src/event.rs

```diff
@@ -135,27 +130,55 @@ enum State {
 ///
 #[derive(Debug)]
 pub struct Events<E: Event> {
-    events_a: Vec<EventInstance<E>>,
-    events_b: Vec<EventInstance<E>>,
-    a_start_event_count: usize,
-    b_start_event_count: usize,
+    // Holds the oldest still active events
+    // Note that a.start_event_count + a.len() should always === events_b.start_event_count
+    events_a: EventSequence<E>,
+    // Holds the newer events
+    events_b: EventSequence<E>,
```

Comment:
```suggestion
    /// Holds the oldest still active events.
    /// Note that a.start_event_count + a.len() should always === events_b.start_event_count.
    events_a: EventSequence<E>,
    /// Holds the newer events.
    events_b: EventSequence<E>,
```

I also added periods at the end of sentences.